### ISSUE TITLE
alternative to astropy.units with pint

### ISSUE THREAD

**Issue by jacopoabramo** [IvoVellekoop/openwfs#31]
I'm currently writing a simulator using openwfs to test my application and generating some synthetic data; I noticed that openwfs uses astropy.units to address the issue of how to manage units and such.

I gave a quick glance online and found [pint](https://github.com/hgrecco/pint). I believe it could be used as a drop-in replacement in some capacity and would remove the need of importing the entire astropy package which - from what I can tell - seems pretty big.

I don't know what happens when installing astropy on itself - if it downloads the entire package or just a subsection of it which does not include optional dependency - but I thought I would just throw this option in the midst if it's of your interest.

**Comment by IvoVellekoop** [comment]
I agree this should be handled in an upcoming release. I plan to add pint as a dependency and remove the astropy dependency, while keeping compatibility with astropy units for users that want to use that package. 
The main problem with astropy is not that it is big, but that all its dependencies cause compatibility issues (I've had problems with astropy requiring numpy 1.x, making the whole of openwfs incompatible with numpy 2.x)

**Comment by jacopoabramo** [comment]
Let me know if I can help with that
